### ISSUE TITLE
geom_col() sometimes doesn't work with dates/numbers in the x axis?

### ISSUE THREAD

**Issue by sowla** [tidyverse/ggplot2#3844]
When it fails, the columns seem to be stacked horizontally and the date seems to be treated as a number. It seems like a bug to me, but if I misunderstood something, I'm sorry and please let me know!

``` r
library(dplyr, warn.conflicts = FALSE)
#> Warning: package 'dplyr' was built under R version 3.6.2
library(ggplot2, warn.conflicts = FALSE)

tibble(
  date = c(Sys.Date(), Sys.Date()-1),
  n = c(1, 2)
) %>% 
  ggplot() + 
  geom_col(aes(date, n))
```

![](https://i.imgur.com/sOqyZih.png)

``` r
# but this doesn't
tibble(
  date = c(Sys.Date(), Sys.Date() -1),
  n = c(1, 1)
) %>% 
  ggplot() + 
  geom_col(aes(date, n))
```

![](https://i.imgur.com/CnWROQp.png)

``` r
# this doesn't
tibble(
  x = c(1, 2),
  n = c(1, 1)
) %>% 
  ggplot() + 
  geom_col(aes(x, n))
```

![](https://i.imgur.com/HQg19iW.png)

``` r
# this does
tibble(
  x = c("a", "b"),
  n = c(1, 1)
) %>% 
  ggplot() + 
  geom_col(aes(x, n))
```

![](https://i.imgur.com/LegfUli.png)

<sup>Created on 2020-02-29 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by yutannihilation** [comment]
Thanks, this is already fixed by #3763 on `v3.3.0-rc` branch, but not on `master`. After version 3.3.0 is released on CRAN, the fix will be merged into master as well.

**Comment by sowla** [comment]
Thank you!! Installing the new version fixes my problem too :)
